File: pickPrime.cpp
#include "pickPrime.h"
largeInt pickPrime(int bits){
    largeInt possiblePrime;
    while(true){
        possiblePrime.digits = randomNumber(bits,true);
        if(!commonPrimeCheck(possiblePrime))continue;
        if(miller_rabin(possiblePrime,40))return possiblePrime;
    }
}

File: pythonAlgos/pickPrime.py
import secrets
from pythonAlgos.miller_rabin import miller_rabin
def pickPrime(bits):
    possiblePrime = secrets.randbits(bits)
    while(not miller_rabin(possiblePrime,40)): 
        
        possiblePrime = secrets.randbits(bits)#just pick another number lol
    return possiblePrime